src/Missle.hpp

```diff
@@ -1,12 +1,13 @@
 #pragma once
 #include <SFML/Graphics.hpp>
 
+#include "MovementState.hpp"
```

Comment:
Reference argument doesn't require to have full definition. You can use forward declaration.

Reply:
It's true, however when I wanted to create a Missle I had to additionally include MovementState in the client file. I thought that maybe it's better to keep them together: whenever you want to create a Missle / Trace you do not have to include both `Missle.hpp` / `Trace.hpp` and `MovementState.hpp`. But I can change it if you think it's better :)

Reply:
No, you have the point :)